packages/schema/src/Schema.ts

```diff
@@ -2244,12 +2244,15 @@ export const optional: {
     self: Schema.All extends S ? "you can't apply optional implicitly, use optional() instead" : S,
     options?: Options
   ): [undefined] extends [Options] ? optional<S> : optionalWithOptions<S, Options>
-} = dual((args) => isSchema(args[0]), (from, options) => {
+} = (dual((args) => isSchema(args[0]), (
+  from,
+  options
+) => {
   // Note: `Schema.All extends S ? "you can't...` is used to prevent the case where `optional` is implicitly applied.
   // For example: `S.String.pipe(S.optional)` would result in `S.String` being inferred as `Schema.All`,
   // which is not the intended behavior. This is mostly an aesthetic consideration, so if it causes issues, we can remove it.
   return new PropertySignatureWithFromImpl(optionalPropertySignatureAST(from, options), from)
-})
+})) as unknown as typeof optional
```

Comment:
Not sure why this is needed, but I had to put it here to make it work. Remember the `dual` application before was inferred as `(...args: any[]) => any` so this stuff is expected on complex cases, still, should be doable without casting.